request.go

```diff
@@ -119,19 +136,19 @@ func (c *Client) createRequest(method, api string, reqbody interface{}) (*http.R
 	}
 
 	// Set the necessary headers
-	req.Header.Add("Content-Type", "application/json")
-	if c.token != nil {
-		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", *c.token))
+	if strings.Contains(api, "JSSResource") || c.token == nil {
+		req.Header.Add("Content-Type", "application/xml")
+		req.SetBasicAuth(c.username, c.password)
```

Comment:
[nits / feature] I don't know much about it, but would it be more secure communicate with them if they were encrypted?

Can we encrypt it?

Reply:
like
```shell
### ref: https://www.jamf.com/developers/apis/classic/code-samples/
$ echo "Basic $(echo -n 'username:password' | base64)"
Basic xxxx

$ curl -H "Accept: application/xml" -H "Authorization: Basic xxxx" https://xxx.jamfcloud.com/JSSResource/policies
```

Reply:
A base64 encoding is just obfuscation. I've got no problem with using it though if we want to make the transition. 

We can't encrypt though.